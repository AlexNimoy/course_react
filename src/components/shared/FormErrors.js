import React from 'react';

const FormErrors = ({ errors, touched, fieldName }) => {
  if(errors[fieldName] && touched[fieldName]) {
    return(
      <div className='error'>
        { errors[fieldName] }
      </div>
    );
  }

  return null;
}

export default FormErrors;
