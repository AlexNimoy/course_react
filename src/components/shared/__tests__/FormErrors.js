import React from 'react';
import { create } from 'react-test-renderer';
import FormErrors from '../FormErrors';

test('return error div', () => {
  const component = create(
    <FormErrors
      errors = {{ fio: 'Required' }}
      touched = {{ fio: true }}
      fieldName = 'fio'
    />
  );

  expect(component.toJSON()).toMatchSnapshot();
})

test('return null if not have errors', () => {
  const component = create(
    <FormErrors
      errors = {{}}
      touched = {{ fio: true }}
      fieldName = 'fio'
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
})

test('return null if exist error for other fields', () => {
  const component = create(
    <FormErrors
      errors = {{ phone: 'Required' }}
      touched = {{ fio: true }}
      fieldName = 'fio'
    />
  );

  expect(component.toJSON()).toMatchSnapshot();
})

test('return null if field not touched', () => {
  const component = create(
    <FormErrors
      errors = {{ fio: 'Required' }}
      touched = {{}}
      fieldName = 'fio'
    />
  );

  expect(component.toJSON()).toMatchSnapshot();
})
