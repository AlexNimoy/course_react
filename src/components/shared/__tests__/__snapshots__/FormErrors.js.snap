// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return error div 1`] = `
<div
  className="error"
>
  Required
</div>
`;

exports[`return null if exist error for other fields 1`] = `null`;

exports[`return null if field not touched 1`] = `null`;

exports[`return null if not have errors 1`] = `null`;
